datarepo-clienttests/src/main/java/testscripts/SoftDeleteDataset.java

```diff
@@ -0,0 +1,215 @@
+
+    BulkLoadFileModel fileLoadModel =
+        new BulkLoadFileModel()
+            .sourcePath(sourceUri.toString())
+            .description("softDel")
+            .mimeType("text/plain")
+            .targetPath(targetPath);
+    String loadTag = FileUtils.randomizeName("softDelTest");
+    BulkLoadArrayRequestModel fileLoadModelArray =
+        new BulkLoadArrayRequestModel()
+            .profileId(datasetSummaryModel.getDefaultProfileId())
+            .loadTag(loadTag)
+            .maxFailedFileLoads(0);
+    fileLoadModelArray.addLoadArrayItem(fileLoadModel);
+
+    JobModel ingestFileJobResponse =
+        repositoryApi.bulkFileLoadArray(datasetSummaryModel.getId(), fileLoadModelArray);
+    ingestFileJobResponse = DataRepoUtils.waitForJobToFinish(repositoryApi, ingestFileJobResponse);
+    BulkLoadArrayResultModel bulkLoadArrayResultModel =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestFileJobResponse, BulkLoadArrayResultModel.class);
+    String fileId = bulkLoadArrayResultModel.getLoadFileResults().get(0).getFileId();
+
+    // ingest the tabular data from the JSON file we just generated
+    // generate a JSON file with the fileref
+    String jsonLine =
+        "{\"VCF_File_Name\":\"name1\", \"Description\":\"description1\", \"VCF_File_Ref\":\""
+            + fileId
+            + "\"}\n";
+    byte[] fileRefBytes = jsonLine.getBytes(StandardCharsets.UTF_8);
+    String jsonFileName = FileUtils.randomizeName("this-better-pass") + ".json";
+    String dirInCloud = "scratch/softDel";
+    String fileRefName = dirInCloud + "/" + jsonFileName;
+    String gsPath = FileUtils.createGsPath(fileRefBytes, fileRefName, testConfigGetIngestbucket);
+
+    IngestRequestModel ingestRequest =
+        new IngestRequestModel()
+            .format(IngestRequestModel.FormatEnum.JSON)
+            .table("vcf_file")
+            .path(gsPath);
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+    logger.info("Successfully loaded data into dataset: {}", ingestResponse.getDataset());
+
+    String datasetId = datasetSummaryModel.getId(); // TODO should these be in utils?
+    DatasetModel datasetModel = repositoryApi.retrieveDataset(datasetId);
+    String dataProject = datasetModel.getDataProject();
+    String tableName = datasetModel.getSchema().getTables().get(0).getName();
+
+    // get row ids for table  TODO combine SQL query w queryForFileRefs in DRSLookup
+    String tableRef =
+        String.format(
+            "`%s.%s.%s`",
+            dataProject,
+            "datarepo_" + datasetModel.getName(), // TODO const ni BQ utils
```

Comment:
Agreed on pulling out anything shared between the DRSLookup and this test into BQ utils.